ctapipe_io_magic/__init__.py

```diff
@@ -889,6 +963,160 @@ def parse_metadata_info(self):
+                laser.CloudLayerAlt = laser_info_runh['MReportLaser.fCloudLayerAlt'][0]
+                laser.CloudLayerDens = laser_info_runh['MReportLaser.fCloudLayerDens'][0]
+                laser.Offset = laser_info_runh['MReportLaser.fOffset'][0]
+                laser.Offset_Calculated = laser_info_runh['MReportLaser.fOffset_Calculated'][0]
+                laser.Offset_Fitted = laser_info_runh['MReportLaser.fOffset_Fitted'][0]
+                laser.Offset2 = laser_info_runh['MReportLaser.fOffset2'][0]
+                laser.Offset3 = laser_info_runh['MReportLaser.fOffset3'][0]
+                laser.Background1 = laser_info_runh['MReportLaser.fBackground1'][0]
+                laser.Background2 = laser_info_runh['MReportLaser.fBackground2'][0]
+                laser.BackgroundErr1 = laser_info_runh['MReportLaser.fBackgroundErr1'][0]
+                laser.BackgroundErr2 = laser_info_runh['MReportLaser.fBackgroundErr2'][0]
+                laser.RangeMax = laser_info_runh['MReportLaser.fRangeMax'][0]
+                laser.RangeMax_Clouds = laser_info_runh['MReportLaser.fRangeMax_Clouds'][0]
+                laser.ErrorCode = laser_info_runh['MReportLaser.fErrorCode'][0]
+                laser.ScaleHeight_fit = laser_info_runh['MReportLaser.fScaleHeight_fit'][0]
+                laser.Alpha_fit = laser_info_runh['MReportLaser.fChi2Alpha_fit'][0]
+                laser.Chi2Alpha_fit = laser_info_runh['MReportLaser.fChi2Alpha_fit'][0]
+                laser.Alpha_firstY = laser_info_runh['MReportLaser.fAlpha_firstY'][0]
+                laser.Alpha_Junge = laser_info_runh['MReportLaser.fAlpha_Junge'][0]
+                laser.PBLHeight = laser_info_runh['MReportLaser.fPBLHeight'][0]
+                laser.Chi2Full_fit = laser_info_runh['MReportLaser.fChi2Full_fit'][0]
+                laser.HWSwitchMaxOffset = laser_info_runh['MReportLaser.fHWSwitchMaxOffset'][0]
+                laser.NCollapse = laser_info_runh['MReportLaser.fNCollapse'][0]
+                laser.Shots = laser_info_runh['MReportLaser.fShots'][0]
+                laser.T0Shift = laser_info_runh['MReportLaser.fT0Shift'][0]
+                laser.Interval_0 = laser_info_runh['MReportLaser.fInterval_0'][0]
+                laser.RCS_min_perfect = laser_info_runh['MReportLaser.fRCS_min_perfect'][0]
+                laser.RCS_min_clouds = laser_info_runh['MReportLaser.fRCS_min_clouds'][0]
+                laser.RCS_min_mol = laser_info_runh['MReportLaser.fRCS_min_mol'][0]
+                laser.LIDAR_ratio = laser_info_runh['MReportLaser.fLIDAR_ratio'][0]
+                laser.LIDAR_ratio_Cloud = laser_info_runh['MReportLaser.fLIDAR_ratio_Cloud'][0]
+                laser.LIDAR_ratio_Junge = laser_info_runh['MReportLaser.fLIDAR_ratio_Junge'][0]
+
+                millisec_seconds = millisec_value * 1e-3
+                combined_mjd_value = mjd_value + millisec_seconds / 86400
+                laser.MJD = combined_mjd_value
```

Comment:
Maybe what can be done here already is to filter for unique reports. In general, there are repeated reports i.e. with the same `fMilliSec` value. The repeated ones should be removed and only one kept for each.

Reply:
Done